### PR TITLE
UNI-190 #comment Make chart fixed-width, so data does not "accordian"…

### PR DESCRIPTION
**Pull request by brev**
UNI-190 #comment Make chart fixed-width, so data does not "accordian" while loading. Went with 200 timepoints for now.

### CODE REVIEW

**Comment by brev** [comment]
ready for review @marionleborgne @lscheinkman @jcasner

File: unicorn/frontend/browser/components/Chart.jsx [hunk 1]
```diff
@@ -51,7 +53,13 @@ export default class Chart extends React.Component {
 
   constructor(props, context) {
     super(props, context);
+
+    // DyGraphs chart container
     this._dygraph = null;
+
+    // Chart Range finder value: For Fixed-width-chart & auto-scroll-to-right
+    this._chartXrange = 200;  // static 200px
+    // this._chartXrange = Math.floor((graphXmax - graphXmin) * 0.2); // dynamic
```
Comment (lscheinkman) [inline unicorn/frontend/browser/components/Chart.jsx]: Did you mean to leave this line commented out here?

**Comment by marionleborgne** [comment]
@brev Just tried it. The UX is much nicer!

**Comment by marionleborgne** [comment]
There is an issue (though I am not sure this is due to your change @brev) that happens when I select multiple metrics at the same time: The graphs get all scrambled like if the timestamps got shuffled. Do you get that too?

**Comment by brev** [comment]
@marionleborgne I've never tried that, I'll look in the morn thanks

File: unicorn/frontend/browser/components/Chart.jsx [hunk 2]
```diff
@@ -64,14 +72,19 @@ export default class Chart extends React.Component {
   componentWillUnmount() {
     if (this._dygraph) {
       this._dygraph.destroy();
-      this._dygraph = null;
+      delete this._dygraph;
     }
+    delete this._chartXrange;
```
Comment (lscheinkman) [inline unicorn/frontend/browser/components/Chart.jsx]: Did you mean to delete the `_chartXrange` property or just reset the value?
See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete

**Comment by lscheinkman** [comment]
:+1: LGTM. It looks great!

File: unicorn/frontend/browser/components/Chart.jsx (hunk 2 above)
Comment (brev) [inline unicorn/frontend/browser/components/Chart.jsx]: @lscheinkman I was treating it as the same, but forgot about the context. Nice catch, thanks.

**Comment by brev** [comment]
merging now, I have issues noted in email to investigate, I'll file jiras where necessary. thanks all.

File: unicorn/frontend/browser/components/Chart.jsx [hunk 3]
```diff
@@ -66,12 +73,19 @@ export default class Chart extends React.Component {
       this._dygraph.destroy();
       this._dygraph = null;
     }
+    if (this._chartXrange) {
+      this._chartXrange = null;
```
Comment (lscheinkman) [inline unicorn/frontend/browser/components/Chart.jsx]: You probably need to initialize the value on `componentDidMount`. You can the `mount/umount` events fired by collapsing the chart on the UI

**Pull request merged**